app/Http/Controllers/BookController.php

```diff
@@ -101,4 +101,40 @@ public function claimById(int|string $id, Request $request)
             'message' => "Book $id was claimed"
         ]);
     }
+
+    public function returnById(int|string $id, Request $request)
+    {
+        $request->validate([
+            'email' => 'required|email'
+        ]);
+
+        $book = Book::find($id);
+
+        if (!$book) {
+            return response()->json([
+                'message' => "Book $id was not found"
+            ], 404);
+        }
+
+        if (!$book->claimed_by_name) {
+            return response()->json([
+                'message' => "Book $id is not currently claimed"
+            ], 400);
+        }
+
+        if ($book->claimed_by_email !== $request->email) {
```

Comment:
This IF block should return the message below. 

Code: 400 BAD REQUEST
Content: `{"message": "Book 1 was not returned. test@test.com did not claim this book."}`

The condition for the 500 error should come after `$book->save()`
`$book->save() ? 200 : 500;`

Reply:
Example:
```
 if ($newReview->save()) {
            return response()->json([
                'message' => 'Review created'
            ], 201);
        } else {
            return response()->json([
                'message' => 'Unexpected error occurred'
            ], 500);
        }
```

Reply:
We were unsure on how to categorise this error from the documentation, but I see what you mean and I agree